packages/proxy-agent/test/test.ts

```diff
@@ -252,9 +254,109 @@ describe('ProxyAgent', () => {
 			);
 		});
 
+		it('should work with `keepAlive: true`', async () => {
+			httpsServer.on('request', function (req, res) {
+				res.end(JSON.stringify(req.headers));
+			});
+
+			process.env.HTTP_PROXY = httpsProxyServerUrl.href;
+			const agent = new ProxyAgent({
+				keepAlive: true,
+				rejectUnauthorized: false,
+			});
+
+			try {
+				const res = await req(new URL('/test', httpsServerUrl), {
+					agent,
+				});
+				res.resume();
+				expect(res.headers.connection).toEqual('keep-alive');
+				expect(res.statusCode).toEqual(200);
+				const s1 = res.socket;
+				await once(s1, 'free');
+
+				const res2 = await req(new URL('/test', httpsServerUrl), {
+					agent,
+				});
+				res2.resume();
+				expect(res2.headers.connection).toEqual('keep-alive');
+				expect(res2.statusCode).toEqual(200);
+				const s2 = res2.socket;
+				assert(s1 === s2);
+
+				await once(s2, 'free');
+			} finally {
+				agent.destroy();
+			}
+		});
+
+		it('should cache agent for multiple http requests', async () => {
+			let gotHttpReq = false;
+			httpServer.on('request', function (req, res) {
+				res.end(JSON.stringify(req.headers));
+				gotHttpReq = true;
+			});
+
+			process.env.ALL_PROXY = httpsProxyServerUrl.href;
+			const agent = new ProxyAgent({ rejectUnauthorized: false });
+
+			const res = await req(httpServerUrl, {
+				agent,
+			});
+			const body = await json(res);
+			assert(gotHttpReq);
+			assert.equal(httpServerUrl.host, body.host);
+			expect(agent.cache.size).toEqual(1);
+			expect([...agent.cache.values()][0]).toBeInstanceOf(HttpProxyAgent);
+
+			gotHttpReq = false;
+			const res2 = await req(httpServerUrl, {
+				agent,
+			});
+			const body2 = await json(res2);
+			assert(gotHttpReq);
+			assert.equal(httpServerUrl.host, body2.host);
+			expect(agent.cache.size).toEqual(1);
+			expect([...agent.cache.values()][0]).toBeInstanceOf(HttpProxyAgent);
+		});
+
+		it('should cache agent for multiple https requests', async () => {
+			let gotHttpsReq = false;
+			httpsServer.on('request', function (req, res) {
+				res.end(JSON.stringify(req.headers));
+				gotHttpsReq = true;
+			});
+
+			process.env.ALL_PROXY = httpsProxyServerUrl.href;
+			const agent = new ProxyAgent({ rejectUnauthorized: false });
+
+			const res = await req(httpsServerUrl, {
+				agent,
+			});
+			const body = await json(res);
+			assert(gotHttpsReq);
+			assert.equal(httpsServerUrl.host, body.host);
+			expect(agent.cache.size).toEqual(1);
+			expect([...agent.cache.values()][0]).toBeInstanceOf(
+				HttpsProxyAgent
+			);
+
+			gotHttpsReq = false;
+			const res2 = await req(httpsServerUrl, {
+				agent,
+			});
+			const body2 = await json(res2);
+			assert(gotHttpsReq);
+			assert.equal(httpsServerUrl.host, body2.host);
+			expect(agent.cache.size).toEqual(1);
+			expect([...agent.cache.values()][0]).toBeInstanceOf(
+				HttpsProxyAgent
+			);
+		});
```

Comment:
These new tests assert that:

1. The agent caching works correctly, and
2. When the `keepAlive` option is _not_ enabled, subsequent requests will work as expected.

Reply:
I removed these two new tests that are unrelated to keepAlive in bac30180bd8e629dba7393968d05786ceede0768, in order to descope this PR